platform/pulpcore/app/viewsets/content.py

```diff
@@ -44,6 +45,28 @@ class ContentViewSet(NamedModelViewSet):
     serializer_class = ContentSerializer
     filter_class = ContentFilter
 
+    @transaction.atomic
+    def create(self, request):
+        """
+        Create a new Content instance from
+        :param request:
+        :return:
+        """
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        artifacts = serializer.validated_data.pop('artifacts')
+        content = serializer.save()
+
+        for relative_path, artifact in artifacts.items():
+            ca = ContentArtifact(artifact=artifact, content=content, relative_path=relative_path)
+            ca.save()
+
+        headers = self.get_success_headers(serializer.data)
+        return Response(serializer.data, status=status.HTTP_201_CREATED, headers=headers)
+
+    def update(self, request, pk):
```

Comment:
All these methods are added by inheriting from a mixin for each. So the solution is to explicitly iherit from the CreateMixin and DeleteMixin.